ApollosProject/apollos-templates
# Updates splash screen package

## DESCRIPTION
<img src="https://user-images.githubusercontent.com/2053547/76439396-ba49df80-6392-11ea-9cc7-fc0211d827cf.gif" width="30%" /><img src="https://user-images.githubusercontent.com/2053547/76439451-c897fb80-6392-11ea-88a1-d05ab539ba2c.gif" width="30%"/>

### What does this PR do, or why is it needed?
This PR switches the package we use to manage splash screens and migrates us away from `react-native-splash-screen` to `react-native-bootsplash`. The old package was abandoned, didn't support rn60 so wasn't fully working on ios, and [an announcement from apple](https://developer.apple.com/news/?id=01132020b) required everyone to change how "launch/splash" screens were implemented. The new package checks all the boxes. 
 
### What design trade-offs/decisions were made?
Some times in the simulator I notice a slight flicker during launch on ios. It's not all the time but does show up in the gif above. The maintainer is tracking this or a similar issue [here](https://github.com/zoontek/react-native-bootsplash/issues/59).

Additionally, this PR has a lot of file changes but its mostly file name changes for image assets and their respective platform references. The actual footprint of this PR is small and if you aren't super familiar with the native implementations for this feature the file changes aren't going to make a ton of sense anyways. You are welcome to look over the packages installation instructions to check the relevant file changes.

### How do I test this PR?
1. Launch app.
2. 👍 

Some things worth considering:
- Android
  - I wasn't completely sure where to put the `launchMode`. Based on the information [here](https://github.com/zoontek/react-native-bootsplash#handle-deep-linking-on-android) I added it to both activities. I didn't, however, add the second `intent-filter` that react-native-bootsplash says you need for deeplinking to work. I think we should test if it works without and then add it if we do. @vinnyjth maybe you will know more about the `android:scheme` it says we'll need for deeplinks.

### What automated tests did you write?
This is all native code for the most part. Nothing to write tests for.

## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
